acceptance/tests/restart_host_run_puppet.rb

```diff
@@ -48,7 +48,14 @@
 
   step "restart each agent" do
     applicable_agents.each do |agent|
-      agent.reboot
+      # Work-around reboot hanging on reply (BKR-971)
+      begin
+        Timeout.timeout(5) do
+          agent.reboot
+        end
```

Comment:
Sleep doesn't work, because it's the `agent.reboot` itself that's hanging as it waits for a response to its shutdown command.